### PR TITLE
quiche: switch external dependency to github

### PR DESCRIPTION
**Pull request by danzh2010**
Switch upstream from https://quiche.googlesource.com/quiche to https://github.com/google/quiche. 
Still the same commit.

### CODE REVIEW

**Comment by moderation** [comment]
Missing replace at https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/http/alternate_protocols_cache/BUILD#L21

**Comment by htuch** [comment]
Awesome! Please fix build, i.e. what @moderation suggests and we can merge.
/wait

**Comment by danzh2010** [comment]
I missed a few places referencing to com_googlesource_quiche. Fixed them now. PTAL

**Review by htuch — APPROVED** [review]
LGTM, thanks! Appreciate the follow-thru and making our supply chain more defensible.

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -804,14 +804,15 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
-    com_googlesource_quiche = dict(
+    com_github_google_quiche = dict(
         project_name = "QUICHE",
         project_desc = "QUICHE (QUIC, HTTP/2, Etc) is Google‘s implementation of QUIC and related protocols",
-        project_url = "https://quiche.googlesource.com/quiche",
+        project_url = "https://github.com/google/quiche",
         version = "29fb6fc501e4df5874129408b5474a91eaa9b7a0",
-        sha256 = "32e7b8ce4e91bb45b617bbf9721132f9218b86d9b3821ebfa32b1c7fc01aafae",
+        sha256 = "bcc17653958c622e475deacbf23e9a215cdb711688671110e20192500ba210dc",
         # Static snapshot of https://quiche.googlesource.com/quiche/+archive/{version}.tar.gz
```
Comment (bencebeky) [inline bazel/repository_locations.bzl:813]: I think this comment line could probably be removed now.

**Pull request merged**